lnwallet/wallet.go

```diff
@@ -1157,6 +1157,27 @@ func (l *LightningWallet) CurrentNumAnchorChans() (int, error) {
 	return numAnchors, nil
 }
 
+// GetWaitingCloseCommitment returns the waiting close channel in case it does
+// exist in the underlying channel state database.
+func (l *LightningWallet) GetWaitingCloseChannel(
+	chanPoint wire.OutPoint) (*channeldb.OpenChannel, error) {
+
+	// Count all anchor channels that are open or pending
+	// open, or waiting close.
+	chans, err := l.Cfg.Database.FetchWaitingCloseChannels()
```

Comment:
IMO this method should be moved to `walletkit_server.go`, and we update the `Config` in the wallet server to take a `db *channeldb.ChannelStateDB`. This way we can avoid passing methods as config values.

Reply:
done